src/main/java/org/folio/event/service/UserTenantService.java

```diff
@@ -51,4 +76,35 @@ public Future<Boolean> deleteUserTenant(UserTenant userTenant, String tenantId,
     PostgresClient pgClient = pgClientFactory.apply(vertx, tenantId);
     return pgClient.withConn(conn -> tenantRepository.deleteUserTenant(conn, userTenant, tenantId));
   }
+
+  public Future<Void> isUsernameUpdatedAndUniqueAcrossTenants(User entity, User userFromStorage, Map<String, String> okapiHeaders, Conn conn, Context vertxContext) {
+    return ObjectUtils.notEqual(entity.getUsername(), userFromStorage.getUsername()) ? isUsernameUniqueAcrossTenants(entity, okapiHeaders, conn, vertxContext) : Future.succeededFuture();
+  }
+
+  public Future<Void> isUsernameUniqueAcrossTenants(User entity, Map<String, String> okapiHeaders, Conn conn, Context vertxContext) {
+    return getConsortiaCentralTenantId(conn, okapiHeaders)
+      .compose(consortiaCentralTenantId -> {
+        if (Objects.nonNull(consortiaCentralTenantId)) {
+          logger.info("Found central tenant id = {}", consortiaCentralTenantId);
+          return isUsernameUniqueAcrossTenants(entity.getUsername(), consortiaCentralTenantId, okapiHeaders, vertxContext);
+        }
+        return Future.succeededFuture();
+      });
+  }
+
+  private Future<Void> isUsernameUniqueAcrossTenants(String username, String consortiaCentralTenantId, Map<String, String> okapiHeaders, Context vertxContext) {
+    Map<String, String> okapiHeadersForCentralTenant = new HashMap<>(okapiHeaders);
+    okapiHeadersForCentralTenant.put(OkapiConnectionParams.OKAPI_TENANT_HEADER, consortiaCentralTenantId);
+    okapiHeadersForCentralTenant.put(RestVerticle.OKAPI_HEADER_TENANT, consortiaCentralTenantId);
+    PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeadersForCentralTenant);
+    return postgresClient.withTrans(conn -> tenantRepository.isUsernameAlreadyExists(conn, username, consortiaCentralTenantId)
```

Comment:
We doing only select action, so don't need to create transaction here,
just use postgresClient.withConn

Reply:
Refactored